dd-sdk-android-ndk/src/main/cpp/utils/backtrace-handler.cpp

```diff
@@ -0,0 +1,78 @@
+#include "backtrace-handler.h"
+
+#include <cstdlib>
+#include <cstdio>
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <iomanip>
+#include <iosfwd>
+#include <unwind.h>
+#include <unistd.h>
+#include <string>
+
+
+static const size_t STACK_SIZE = 30;
+
+struct BacktraceState {
+    uintptr_t *current;
+    uintptr_t *end;
+};
+
+namespace {
+    _Unwind_Reason_Code unwind_callback(struct _Unwind_Context *context, void *arg) {
+        auto *state = static_cast<BacktraceState *>(arg);
+        uintptr_t pc = _Unwind_GetIP(context);
+        if (pc) {
+            if (state->current == state->end) {
+                return _URC_END_OF_STACK;
+            } else {
+                *state->current++ = pc;
+            }
+        }
+        return _URC_NO_REASON;
+    }
+
+
+    size_t capture_backtrace(uintptr_t *buffer, size_t max) {
+        BacktraceState state = {buffer, buffer + max};
+        _Unwind_Backtrace(unwind_callback, &state);
+        return state.current - buffer;
+    }
+
+    const char *to_hexa(char *buffer_pointer, uintptr_t pointer) {
+        std::sprintf(buffer_pointer, "0x%x", pointer);
+        return buffer_pointer;
+    }
+
+}
+
+namespace trace {
+    const char *generate_backtrace() {
+        uintptr_t buffer[STACK_SIZE];
+        std::string backtrace;
+        size_t captured_stacksize = capture_backtrace(buffer, STACK_SIZE);
+        for (size_t idx = 0; idx < captured_stacksize; ++idx) {
+            void *addr = reinterpret_cast<void *>(buffer[idx]);
```

Comment:
let's use `uintptr_t` instead of `void *` since this is how you have declared `buffer[]`

Reply:
I would...but the below method `dladdr(addr, &info)` accepts a `void *` as a param :(